From ce630fe5c77b558adba701c7045406cd61bbb2f9 Mon Sep 17 00:00:00 2001
From: Jordan Hand <jorhand@microsoft.com>
Date: Wed, 8 Jan 2020 13:27:28 -0800
Subject: [PATCH] Mark setjmp as returns_twice

setjmp has returns_twice behavior that prevents the compiler from being
able to reason about the control flow behavior of the function. Mark it
with the returns_twice attribute so that the compiler knows it is not a
normal function.

Signed-off-by: Jordan Hand <jorhand@microsoft.com>
---
 3rdparty/musl/musl/include/setjmp.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/3rdparty/musl/musl/include/setjmp.h b/3rdparty/musl/musl/include/setjmp.h
index 2d43abf84..1e24de761 100644
--- a/3rdparty/musl/musl/include/setjmp.h
+++ b/3rdparty/musl/musl/include/setjmp.h
@@ -19,17 +19,17 @@ typedef struct __jmp_buf_tag {
  || defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) \
  || defined(_BSD_SOURCE)
 typedef jmp_buf sigjmp_buf;
-int sigsetjmp (sigjmp_buf, int);
+int sigsetjmp (sigjmp_buf, int) __attribute__((returns_twice));
 _Noreturn void siglongjmp (sigjmp_buf, int);
 #endif
 
 #if defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) \
  || defined(_BSD_SOURCE)
-int _setjmp (jmp_buf);
+int _setjmp (jmp_buf) __attribute__((returns_twice));
 _Noreturn void _longjmp (jmp_buf, int);
 #endif
 
-int setjmp (jmp_buf);
+int setjmp (jmp_buf) __attribute__((returns_twice));
 _Noreturn void longjmp (jmp_buf, int);
 
 #define setjmp setjmp
-- 
2.17.1

